### DISCUSSION TITLE
Best practice for making the Web UI secure, accessible via https?

### DISCUSSION THREAD

**Discussion by MegalithicBTC** [wg-easy/wg-easy#987] [General]
Thanks for this great project, which I was able to get running easily. Documentation seems great.

I do have a blocking issue with regard to the Web UI.

Using the documentation, it seems by default the Web UI is not secure:

![not-secure](https://github.com/wg-easy/wg-easy/assets/162853102/a45a9f8d-98ff-4197-8e4a-dcc8e146e3a3)

My understanding is that it might therefore be unsafe to use the Web UI when the `wg-easy` is running on a remote server.

For example, if I were to enter my password, or download a configuration file, that data would not be protected by SSL and would be open to a man-in-the-middle attack.

Am I wrong about this?

If I'm not wrong, what would you recommend as the simplest solution?

I looked into running Caddy as a container alongside `wg-easy`, but to provide SSL termination, wouldn't I need a domain name? 

I'm surprised that this issue hasn't come up and there isn't an explicit recommendation on how to use the Web UI in a secure manner. But maybe I'm confused about something.

Another possible workaround - disable the Web UI and somehow get the Client(s) configuration details directly from the command line?

thanks!

**Comment by eandersons** [comment]
Answer to your question can be found in #830. In short - use a reverse proxy by your liking.
certbot can be used to get and automatically renew certificates.

**Comment by MegalithicBTC** [comment]
Thanks. I'm assuming that many people have taken this path before so was hoping someone might have a sample Docker Compose or something.

**Reply by eandersons** [reply]
Try to search for "Let's Encrypt", "certbot", "SSL" or "certificate" in discussions and Wiki, you will definitely find some pointers.

**Comment by MegalithicBTC** [comment]
Oh, thanks, should have found that before. Looks like all the examples require a domain name, so I'll just do that: https://github.com/wg-easy/wg-easy/wiki/Using-WireGuard-Easy-with-Caddy-SSL
